x-pack/plugins/lists/server/routes/create_list_index_route.ts

```diff
@@ -33,33 +33,43 @@ export const createListIndexRoute = (router: ListsPluginRouter): void => {
         const listIndexExists = await lists.getListIndexExists();
         const listItemIndexExists = await lists.getListItemIndexExists();
 
+        try {
+          // Check if the old legacy lists and items template exists and remove it
+          await lists.deleteLegacyListBootStrapIndex();
```

Comment:
We should delete the legacy templates only after the new templates are created below to avoid having a period between deleting the old templates and creating the new ones where there are no templates. During that time, the indices could potentially rollover and there would be no template for the new index so it would be created without mappings or the proper settings.

The "new" style templates override legacy templates if both exist and match an index pattern so it's fine to have both (even with the same name, they're completely separate in that sense) until we delete the legacy templates.